controllers/user/UserController.php

```diff
@@ -98,17 +107,92 @@ public function actionCreate(){
     }
 
     /**
-     * Authenticate the user by username and password
-     * If, the credentials are correct then token will be generated by JWT
+     * Store the user details like email, token in user_invite table
+     * Generate six digit random number code
+     * @param $email
+     * @return array|false
+     * @throws \Throwable
+     * @throws \yii\db\StaleObjectException
+     */
+    public function inviteUser($email){
+        $invite = new Invite();
+        $invite->scenario = 'invite';
+
+        //If the site in maintenance mode
+        if (Yii::$app->settings->get('maintenanceMode')) {
+            Yii::$app->getView()->warn(ControllerAccess::getMaintenanceModeWarningText());
+            return false;
+        }
+
+        $invite->source = Invite::SOURCE_SELF;
+        $invite->language = Yii::$app->language;
+        $invite->email = $email;
+
+        // Delete existing invite for e-mail - but reuse token
+        $existingInvite = Invite::findOne(['email' => $email]);
+        if ($existingInvite !== null) {
+            $invite->token = $existingInvite->token;
+            $existingInvite->delete();
+        }
+
+        $invite->token = $this->generateCode();
+        $checkCodeAlredyExist = Invite::find()->where(['token'=>$invite->token])->one();
+
+        if($checkCodeAlredyExist!=null){
+            $invite->token = $this->generateCode();
+        }
+
+        //We set validation false because we are not fill recaptcha value in API
+        if($invite->save(false)){
+            $this->sendMail($invite);
+            return $this->returnSuccess("Registration successfully done, Please check your email, Code is send to your email address ".$email,200);
+        }
+        return $this->returnError(400,$invite->errors);
+
+    }
+
+    public function generateCode()
+    {
+        return random_int(100000, 999999);
+    }
+
+    /**
+     * Send code to user email address
+     * @param $invite
+     * @return void
+     * @throws \yii\base\Exception
+     * @throws \yii\base\InvalidConfigException
+     */
+    public function sendMail($invite)
+    {
+
+        /** @var Module $module */
+        $module = Yii::$app->moduleManager->getModule('user');
+
+        // User requested registration link by its self
+        if ($invite->source == Invite::SOURCE_SELF) {
+            $mail = Yii::$app->mailer->compose([
+                'html' => '@humhub/modules/smartVillageAPI/views/email_template/UserInviteSelf',
+                'text' => '@humhub/modules/smartVillageAPI/views/email_template/plain_text/UserInviteSelf'
```

Comment:
we get the following error when testing:

<img width="610" alt="Bildschirmfoto 2022-06-07 um 18 41 13" src="https://user-images.githubusercontent.com/1942953/172436730-d4506e4f-9503-438c-a9c3-d707897f3865.png">

our module is called "smart-village-app-humhub-ext", so i think the `smartVillageAPI` needs to be replaced by this, right?

Reply:
@gaurav-handysolver do you change the path for this PR? or could this path be more relative?

Reply:
Hi @donni106 
Which path you are talking ? path of file from smartVillageAPI to smart-village-app-humhub-ext?

Reply:
yes. if it needs to be hardcoded and could not be relative somehow, then this change is necessary.

Reply:
Hi @donni106 
Done!